src/Components/MemberLogged/SubscribeActivities/FormMemberSubscription/index.jsx

```diff
@@ -0,0 +1,129 @@
+  const member = useSelector((state) => state.loginMembers.data);
+  const { success } = useSelector((state) => state.subscriptions);
+  const dispatch = useDispatch();
+
+  const [showConfirmModal, setShowConfirmModal] = useState(false);
+  const [classes, setClasses] = useState([]);
+  const {
+    handleSubmit,
+    register,
+    formState: { errors }
+  } = useForm({
+    mode: 'onChange',
+    resolver: joiResolver(subscriptionSchema),
+    defaultValues: {
+      members: member._id,
+      classes: ''
+    }
+  });
+
+  useEffect(() => {
+    getSubscriptions(dispatch);
+    getClasses(dispatch).then((data) => {
+      filterClassAndOrder(data);
+    });
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (success) {
+      history.push('/user/members/subscribe-class');
+      dispatch(resetState());
+    }
+  }, [success]);
+
+  const filterClassAndOrder = (data) => {
+    const filteredClasses = data.filter(
+      (item) => !item.deleted && item.activity && item.activity._id === id
+    );
+    filteredClasses.sort((a, b) => {
+      const dayComparison = a.day.localeCompare(b.day);
+      if (dayComparison !== 0) {
+        return dayComparison;
+      }
+      return a.time.localeCompare(b.time);
+    });
+    setClasses(filteredClasses);
+  };
+
+  const handleShowConfirmModal = async () => {
+    if (!showConfirmModal) {
+      setShowConfirmModal(true);
+    } else {
+      setShowConfirmModal(false);
+    }
+  };
+  const onSubmit = () => {
+    handleShowConfirmModal();
+  };
+  const onConfirm = (data) => {
+    try {
+      addSubscriptions(dispatch, data);
+      setShowConfirmModal(false);
+    } catch (error) {
+      throw new Error(error);
+    }
+  };
+
+  return (
+    <>
+      {isPending ? (
+        <div className={styles.containerLoader}>
+          <Loader />
+        </div>
+      ) : (
+        <div className={styles.formContainer}>
+          <h2>
+            {classes.length !== 0 ? classes[0].activity.name : 'This activity has no classes'}
+          </h2>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <select name="classes" {...register('classes')}>
+              <option value="">Select a class</option>
+              {classes.map((item) => (
+                <option key={item._id} value={item._id}>
+                  {item.day}/ {item.time}
+                </option>
+              ))}
+            </select>
+            {errors.classes && <p className={styles.error}>Please select a class</p>}
+            <div className={styles.buttonContainer}>
+              <Link to="/user/members/subscribe-class">
+                <Button classNameButton={'cancelButton'} text={'Cancel'} />
+              </Link>
+              {classes.length !== 0 && <Button text={'Submit'} classNameButton={'submitButton'} />}
+            </div>
+          </form>
+          {showConfirmModal && (
+            <ConfirmModal
+              handler={() => handleShowConfirmModal()}
```

Comment:
In the "handler" attribute of "ConfirmModal", instead of using an arrow function, it would be simpler and clearer to directly pass the reference of the "handleShowConfirmModal" function. This will simplify the code and maintain the desired functionality.
Using an anonimous arrow function in this case can create some unnecessary overhead. By directly passing the function reference, we avoid creating an additional function and improve code readability.
```suggestion
              handler={handleShowConfirmModal}
```